composer.json

```diff
@@ -24,7 +24,7 @@
         "php": "^8.0",
         "laudis/typed-enum": "^1.3.2",
         "php-http/discovery": "^1.13",
-        "psr/http-message": "^1.0",
+        "psr/http-message": "^1.0|^2.0",
```

Comment:
Does it need to keep the old version?

Reply:
I'd say so yes to maintain backwards compatibility

Reply:
your project is php^8 and with the change 1.1 to 2.0 ( https://github.com/php-fig/http-message/compare/1.1...2.0 ) you have nothing to worry about. It's just added argument types and return types.